src/components/auth/login-form/LoginForm.tsx

```diff
@@ -0,0 +1,116 @@
+import './LoginForm.scss';
+
+import { LoginRequest } from '../../../types/auth/login-request.type';
+import { PRODUCTS_URL_PREFIX } from '../../../constants/api.constants';
+import React from 'react';
+import { setAxiosAccessToken } from '../../../utils/request.utils';
+import { useAuthContext } from '../../../context/AuthContext';
+import { useForm } from 'react-hook-form';
+import { useLogin } from '../../../hooks/auth/useLogin';
+import { useNavigate } from 'react-router-dom';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+const loginSchema = z.object({
+  emailAddress: z
+    .string()
+    .min(3, 'Email is required')
+    .email('Entered email is invalid.'),
+  password: z.string().min(3, 'Password is required'),
+});
+
+const LoginForm: React.FC = () => {
+  const navigate = useNavigate();
+  const { register, handleSubmit, formState } = useForm<LoginRequest>({
+    resolver: zodResolver(loginSchema),
+    mode: 'onTouched',
+  });
+
+  const { login: authLogin, accessToken } = useAuthContext();
+  const { login, isLoading } = useLogin();
+
+  const handleLogin = async (credentials: LoginRequest) => {
+    const response = await login(credentials);
+    if (response) {
+      authLogin(
+        response.tokens.accessToken,
+        response.tokens.refreshToken,
+        response.customer
+      );
+    }
+
+    if (accessToken) {
+      setAxiosAccessToken(accessToken);
+      navigate(PRODUCTS_URL_PREFIX);
+    }
+  };
+
+  return (
+    <form
```

Comment:
**suggestion:** Keep in mind in the future to separate component into containers and presentational.
Link: https://www.patterns.dev/react/presentational-container-pattern/

![image](https://github.com/user-attachments/assets/8d0b8d9b-aac1-409b-8df9-ff90f8e6e254)

Reply:
Yup, the conversation + the article was really helpful in answering the questions I had regarding best-practices / handling the architecture in a cleaner, more maintainable manner. Keeping this in mind from now on.

Also, I do appreciate that youve put the diagram we discussed upon in here 😄